### DISCUSSION TITLE
Some compatible questions when I use ROCm on AMD GPU.

### DISCUSSION THREAD

**Discussion by Looong01** [pyg-team/pytorch_geometric#6370] [Q&A] [answered]
I have an AMD GPU. I have installed the PyTorch ROCm version and PYG CPU version.
Can I use both of them to build, train and test my GNN model at the same time?
If it will give GPU errors when I use them?
Will the PyTorch ROCm version and PYG CPU version be compatible?

**Comment by rusty1s** [comment]
We added ROCm support to `torch-scatter` and `torch-sparse` lately, so you should find some success installing them via
```
pip install --verbose torch-scatter torch-sparse
```
Let me know if this works.

**Reply by Looong01** [reply]
I can install these libraries definitely. But I do not know how to use it right now.
I want to ask if PYG will work on AMD GPU if I install the latest version of torch-scatter and torch-sparse?
And which version of PYG should I install in ROCm environment? CPU version or CUDA version?

**Reply by rusty1s** [reply]
Yes, it should work on AMD GPU as far as I know if you run
```
pip install torch-scatter torch-sparse torch-geometric

**Reply by Looong01** [reply]
Thank you very much! I will try all the features of PYG on ROCm as far as I can.

**Reply by Looong01** [reply]
> Yes, it should work on AMD GPU as far as I know if you run
> 
> ```
> pip install torch-scatter torch-sparse torch-geometric
> ```

I noticed that the pip installing command on your website (https://pytorch-geometric.readthedocs.io/en/latest/notes/installation.html) is "pip install pyg-lib torch-scatter torch-sparse torch-cluster torch-spline-conv torch-geometric -f https://data.pyg.org/whl/torch-1.13.0+cu117.html", not the command you said above, "pip install torch-scatter torch-sparse torch-geometric".

So, I have some questions:

1. The source url does not include the wheels of ROCM build, right? So, I can not install it by using "-f" from "data.pyg.org/whl", right?

2. I noticed that the packages on PYPI are all sources (tar.gz), not wheels. So the reason why I cannot install them from "data.pyg.org/whl" is because pip will install them from source and only source has ROCm build, while wheels are not, right?

3. "pyg-lib, torch-cluster and torch-spline-conv" are not included in the command you gave above. So, I search them in your repos. I noticed that you have merged the ROCm build of "torch-cluster and torch-spline-conv" into master. But you do not release them to PYPI. So, How can I install the ROCm version of "torch-cluster and torch-spline-conv"? Build them from Github?

4. If I only install "torch-scatter torch-sparse torch-geometric", can I use PYG on AMD GPU successfully? Is there any influence of the absence of the packages "pyg-lib, torch-cluster and torch-spline-conv"?

5. I also cannot install ROCm version from conda, right? Because conda install them from wheels.

**Reply by rusty1s** [reply]
1. We do not provide ROCM builds, so the `-f` option does noting/needs to be dropped. You need to install from source.
2. Yes.
3. You can run
```
pip install git+https://github.com/rusty1s/pytorch_spline_conv.git
```
4. Yes, all other dependencies are optional, e.g., `pytorch-spline-conv` is only needed if you make use of `nn.SplineConv`.
5. No versions for ROCM available on conda either :(

**Reply by Looong01** [reply]
> 1. We do not provide ROCM builds, so the `-f` option does noting/needs to be dropped. You need to install from source.
> 2. Yes.
> 3. You can run
> 
> ```
> pip install git+https://github.com/rusty1s/pytorch_spline_conv.git
> ```
> 
> 4. Yes, all other dependencies are optional, e.g., `pytorch-spline-conv` is only needed if you make use of `nn.SplineConv`.
> 5. No versions for ROCM available on conda either :(

Thank you very much! I will buy an AMD GPU recently. If there are any other questions, I will tell you.

**Reply by Looong01** [reply]
> We added ROCm support to `torch-scatter` and `torch-sparse` lately, so you should find some success installing them via
> 
> ```
> pip install --verbose torch-scatter torch-sparse
> ```
> 
> Let me know if this works.

Thank you very much! I have buy my AMD GPU and fully installed ROCm 5.4.2 stacks and PyTorch 2.0.0.dev20230125+rocm5.3.

But when I use "pip install torch-scatter torch-sparse torch-geometric", it returns:
```
In file included from csrc/hip/segment_csr_hip.hip:3:
      In file included from csrc/hip/../hip/segment_csr_hip.h:4:
      In file included from csrc/hip/../extensions.h:2:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/api/include/torch/torch.h:3:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/api/include/torch/all.h:7:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/api/include/torch/autograd.h:3:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/autograd/autograd.h:3:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/autograd/variable.h:6:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/autograd/cpp_hook.h:2:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/torch/csrc/autograd/function_hook.h:3:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/Tensor.h:3:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/Tensor.h:3:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/TensorBody.h:16:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/c10/core/Scalar.h:11:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/c10/core/ScalarType.h:5:
      In file included from /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/c10/util/Half.h:15:
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/c10/util/complex.h:8:10: fatal error: 'thrust/complex.h' file not found
      #include <thrust/complex.h>
               ^~~~~~~~~~~~~~~~~~
      29 warnings and 1 error generated when compiling for gfx1030.
      error: command '/opt/rocm-5.4.2/bin/hipcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
```
The screenshot:
![image](https://user-images.githubusercontent.com/69568351/214805948-383150f2-1a27-4ecd-a496-0e5ae9c7d083.png)

And torch-sparse, torch-geometric, pytorch_spline_conv and pytorch_cluster have the same problem.
Before that, I met a problem that "\<cmath\> file not found". And After I used "sudo apt install g++-12", it was solved.

**Reply by Looong01** [reply]
Although my torch version is 2.0.0, I think it's not the problem of PyTorch. I think it's of gcc or g++ environment.

**Reply by rusty1s** [reply]
Hopefully @jeffdaily has some insights.

**Reply by Looong01** [reply]
> Hopefully @jeffdaily has some insights.

I have solved the previous problems by installing these two packages (https://github.com/ROCmSoftwarePlatform/rocThrust) and (https://github.com/ROCmSoftwarePlatform/hipSPARSE).

However, it gives another error when it's during the compiling process of "pip install torch-scatter torch-sparse torch-geometric":

```
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:455:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(c10::RegisterOperators::Options&&) &&’
        455 |   RegisterOperators&& op(Options&& options) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:455:23: note:   candidate expects 1 argument, 2 provided
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:461:22: note: candidate: ‘c10::RegisterOperators& c10::RegisterOperators::op(c10::RegisterOperators::Options&&) &’
        461 |   RegisterOperators& op(Options&& options) & {
            |                      ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:461:22: note:   candidate expects 1 argument, 2 provided
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:471:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(const string&, c10::RegisterOperators::Options&&) &&’
        471 |   RegisterOperators&& op(const std::string& schemaOrName, Options&& options = RegisterOperators::options()) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:471:69: note:   no known conversion for argument 2 from ‘int64_t (*)() noexcept’ {aka ‘long int (*)() noexcept’} to ‘c10::RegisterOperators::Options&&’
        471 |   RegisterOperators&& op(const std::string& schemaOrName, Options&& options = RegisterOperators::options()) && {
            |                                                           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:476:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(c10::FunctionSchema, c10::RegisterOperators::Options&&) &&’
        476 |   RegisterOperators&& op(FunctionSchema schema, Options&& options) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:476:41: note:   no known conversion for argument 1 from ‘const char [28]’ to ‘c10::FunctionSchema’
        476 |   RegisterOperators&& op(FunctionSchema schema, Options&& options) && {
            |                          ~~~~~~~~~~~~~~~^~~~~~
      csrc/version.cpp:41:13: warning: ‘registry’ defined but not used [-Wunused-variable]
         41 | static auto registry = torch::RegisterOperators().op(
            |             ^~~~~~~~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
```

The screenshot:
![image](https://user-images.githubusercontent.com/69568351/214896145-b38c16d1-8d28-44b4-9e36-247b67394a43.png)

**Reply by businessabstraction** [reply]
> ROCm

Will you be providing ROCm builds in the future? I am building a box specifically for PyG, andf AMD offers 24Gb for half the price of NVidia.

**Reply by Looong01** [reply]
> > ROCm
> 
> Will you be providing ROCm builds in the future? I am building a box specifically for PyG, andf AMD offers 24Gb for half the price of NVidia.

here, https://github.com/Looong01/pyg-rocm-build

**Reply by Andy3117006664** [reply]
You are my hero

**Reply by yyou1996** [reply]
You are the true hero sir!

**Comment by jeffdaily** [comment]
We don't control these packages and how the wheels are prepared and delivered -- there are no ROCm versions of these wheels that can be pip installed.  You must build all of the prereqs from source.  Be careful with using pip in case it installs a CUDA version of the wheel instead.

The missing thrust header error needs some triage.  The file csrc/hip/segment_csr_hip.hip appears to have been hipified, but didn't convert the thrust include to a hipthrust include.  I'll need to reproduce these steps to confirm what's going wrong.

**Reply by Looong01** [reply]
> We don't control these packages and how the wheels are prepared and delivered -- there are no ROCm versions of these wheels that can be pip installed. You must build all of the prereqs from source. Be careful with using pip in case it installs a CUDA version of the wheel instead.
> 
> The missing thrust header error needs some triage. The file csrc/hip/segment_csr_hip.hip appears to have been hipified, but didn't convert the thrust include to a hipthrust include. I'll need to reproduce these steps to confirm what's going wrong.

Thank you! I have solved the previous problems by installing these two packages (https://github.com/ROCmSoftwarePlatform/rocThrust) and (https://github.com/ROCmSoftwarePlatform/hipSPARSE).

However, it gives another error when it's during the compiling process of "pip install torch-scatter torch-sparse torch-geometric":

```
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:455:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(c10::RegisterOperators::Options&&) &&’
        455 |   RegisterOperators&& op(Options&& options) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:455:23: note:   candidate expects 1 argument, 2 provided
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:461:22: note: candidate: ‘c10::RegisterOperators& c10::RegisterOperators::op(c10::RegisterOperators::Options&&) &’
        461 |   RegisterOperators& op(Options&& options) & {
            |                      ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:461:22: note:   candidate expects 1 argument, 2 provided
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:471:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(const string&, c10::RegisterOperators::Options&&) &&’
        471 |   RegisterOperators&& op(const std::string& schemaOrName, Options&& options = RegisterOperators::options()) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:471:69: note:   no known conversion for argument 2 from ‘int64_t (*)() noexcept’ {aka ‘long int (*)() noexcept’} to ‘c10::RegisterOperators::Options&&’
        471 |   RegisterOperators&& op(const std::string& schemaOrName, Options&& options = RegisterOperators::options()) && {
            |                                                           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:476:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(c10::FunctionSchema, c10::RegisterOperators::Options&&) &&’
        476 |   RegisterOperators&& op(FunctionSchema schema, Options&& options) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:476:41: note:   no known conversion for argument 1 from ‘const char [28]’ to ‘c10::FunctionSchema’
        476 |   RegisterOperators&& op(FunctionSchema schema, Options&& options) && {
            |                          ~~~~~~~~~~~~~~~^~~~~~
      csrc/version.cpp:41:13: warning: ‘registry’ defined but not used [-Wunused-variable]
         41 | static auto registry = torch::RegisterOperators().op(
            |             ^~~~~~~~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
```

The screenshot:
![image](https://user-images.githubusercontent.com/69568351/214896145-b38c16d1-8d28-44b4-9e36-247b67394a43.png)

**Reply by Looong01** [reply]
It seems that it can include the "thrust/complex.h" now. But it gives another error during compiling.

**Reply by Looong01** [reply]
And this:
```
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:455:23: note:   candidate expects 1 argument, 2 provided
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:461:22: note: candidate: ‘c10::RegisterOperators& c10::RegisterOperators::op(c10::RegisterOperators::Options&&) &’
        461 |   RegisterOperators& op(Options&& options) & {
            |                      ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:461:22: note:   candidate expects 1 argument, 2 provided
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:471:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(const string&, c10::RegisterOperators::Options&&) &&’
        471 |   RegisterOperators&& op(const std::string& schemaOrName, Options&& options = RegisterOperators::options()) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:471:69: note:   no known conversion for argument 2 from ‘int64_t (*)() noexcept’ {aka ‘long int (*)() noexcept’} to ‘c10::RegisterOperators::Options&&’
        471 |   RegisterOperators&& op(const std::string& schemaOrName, Options&& options = RegisterOperators::options()) && {
            |                                                           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:476:23: note: candidate: ‘c10::RegisterOperators&& c10::RegisterOperators::op(c10::FunctionSchema, c10::RegisterOperators::Options&&) &&’
        476 |   RegisterOperators&& op(FunctionSchema schema, Options&& options) && {
            |                       ^~
      /home/loong/anaconda3/envs/PyTorch/lib/python3.10/site-packages/torch/include/ATen/core/op_registration/op_registration.h:476:41: note:   no known conversion for argument 1 from ‘const char [28]’ to ‘c10::FunctionSchema’
        476 |   RegisterOperators&& op(FunctionSchema schema, Options&& options) && {
            |                          ~~~~~~~~~~~~~~~^~~~~~
      csrc/version.cpp:41:13: warning: ‘registry’ defined but not used [-Wunused-variable]
         41 | static auto registry = torch::RegisterOperators().op(
            |             ^~~~~~~~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> torch-scatter

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```
![image](https://user-images.githubusercontent.com/69568351/214900889-d8d3c932-7939-4daf-8a97-b1452757edb4.png)

**Reply by Looong01** [reply]
> We don't control these packages and how the wheels are prepared and delivered -- there are no ROCm versions of these wheels that can be pip installed. You must build all of the prereqs from source. Be careful with using pip in case it installs a CUDA version of the wheel instead.
> 
> The missing thrust header error needs some triage. The file csrc/hip/segment_csr_hip.hip appears to have been hipified, but didn't convert the thrust include to a hipthrust include. I'll need to reproduce these steps to confirm what's going wrong.

My environment is:
OS: Ubuntu 22.04.1     Linux home 5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
Python: 3.10
ROCm: 5.4.2
PyTorch: 2.0.0.dev20230125+rocm5.3
GPU: 6600xt (I use ```HSA_OVERRIDE_GFX_VERSION=10.3.0``` to override the GFX name.)

**Comment by jeffdaily** [comment, marked as answer]
inside docker container based on image rocm/pytorch:latest

```
git clone https://github.com/pyg-team/pytorch_geometric
git clone --recursive https://github.com/rusty1s/pytorch_sparse
git clone https://github.com/rusty1s/pytorch_scatter
cd pytorch_sparse
pip install . -vvv |& tee ~/build_sparse.log
cd ../pytorch_scatter
pip install . -vvv |& tee ~/build_scatter.log
cd ../pytorch_geometric
pip install . -vvv |& tee ~/build_geometric.log
```

This was ubuntu 20, rocm 5.4.0.

**Reply by jeffdaily** [reply]
The sparse build takes a long time.  It was single-threaded for me.  But I didn't encounter any build errors in any package.  I can provide log files if needed.  This was building for gfx906 (MI50).

**Reply by Looong01** [reply]
> The sparse build takes a long time. It was single-threaded for me. But I didn't encounter any build errors in any package. I can provide log files if needed. This was building for gfx906 (MI50).

Could you help to provide the wheels you have already built? I can try to install .whl directly without compiling. Maybe that will work. Thank you very much!

**Reply by Looong01** [reply]
Thank you, I have already install these packages by using your commands above.
I think it's the problem of the tar packages on PyPi. 

If I use ```pip install torch-scatter torch-sparse torch-geometric```, it will returns error. But if I git clone them from github repos, it will be successful.

@rusty1s Hey, bro. Maybe some mistakes in the tar packages on PyPi. The codes of Github repos have no problem.
And I have compiled the wheels of ROCm version of ```pytorch_cluster  pytorch_geometric  pytorch_scatter  pytorch_sparse  pytorch_spline_conv```. Do you think it's good idea to release them on your website (https://data.pyg.org/whl/) ?

**Comment by Looong01** [comment]
@rusty1s These are the wheels I built for PyTorch 1.13 from py3.7 to py3.10.
https://github.com/Looong01/pyg-rocm-wheels

**Reply by PavanKumarMiriyala** [reply]
@Looong01 Can PyG be supported on AMD GPUs now?  How can we install pyg-rocm-wheels?

**Reply by Looong01** [reply]
It's supported now. You can download them here https://github.com/Looong01/pyg-rocm-build/releases/

You must be sure that you have already installed ROCm 5.2 or newer versions, and PyTorch 1.13 or newer versions.
Then unzip all these files in a folder, and use ```cd <the folder path>```.
Then enter your target envs, and ```pip install <the wheel's name>``` to install it.

**Reply by Looong01** [reply]
> @Looong01 Can PyG be supported on AMD GPUs now? How can we install pyg-rocm-wheels?

I have updated the README.md here with installation steps.
https://github.com/Looong01/pyg-rocm-wheels

**Reply by PavanKumarMiriyala** [reply]
@Looong01 Awesome, really appreciate your effort. Is ubuntu 22.04 a compulsory requirement? or 20.04 also works?

**Reply by Looong01** [reply]
> @Looong01 Awesome, really appreciate your effort. Is ubuntu 22.04 a compulsory requirement? or 20.04 also works?

I think it's not. It's up to your ROCm version. For example, ROCm 5.2.x supports 18.04 or newer, but ROCm 5.3.x & 5.4.x support 20.04 or newer.

**Reply by PavanKumarMiriyala** [reply]
I'm using ubuntu 20.04, ROCm 5.4.0, Python 3.8.13, torch 1.13.0. But I'm facing the below error

OSError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/conda/lib/python3.8/site-packages/torch_scatter/_version_cuda.so)

**Reply by Looong01** [reply]
> I'm using ubuntu 20.04, ROCm 5.4.0, Python 3.8.13, torch 1.13.0. But I'm facing the below error
> 
> OSError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/conda/lib/python3.8/site-packages/torch_scatter/_version_cuda.so)

What operation did you do before this error?

**Reply by PavanKumarMiriyala** [reply]
I did 

1. wget https://github.com/Looong01/pyg-rocm-build/releases/download/1/torch-1.13-rocm-5.4.2-py38-linux_x86_64.zip
2. unzip torch-1.13-rocm-5.4.2-py38-linux_x86_64.zip
3. pip3 install *.whl

Installation is successful. I'm facing the above error when I run an example gcn code.

**Reply by Looong01** [reply]
> I'm using ubuntu 20.04, ROCm 5.4.0, Python 3.8.13, torch 1.13.0. But I'm facing the below error
> 
> OSError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/conda/lib/python3.8/site-packages/torch_scatter/_version_cuda.so)

You can make an issue here. https://github.com/Looong01/pyg-rocm-build/issues

**Reply by Looong01** [reply]
> I did
> 
> 1. wget https://github.com/Looong01/pyg-rocm-build/releases/download/1/torch-1.13-rocm-5.4.2-py38-linux_x86_64.zip
> 2. unzip torch-1.13-rocm-5.4.2-py38-linux_x86_64.zip
> 3. pip3 install *.whl
> 
> Installation is successful. I'm facing the above error when I run an example gcn code.

I think the libc version in your os is too old.

Solution:
1. ```sudo strings /lib/x86_64-linux-gnu/libc.so.6 |grep GLIBC```, make sure there are no strings include ```GLIBC_2.32```
2. If 1 is true, download th newest or required version of glibc (http://ftp.gnu.org/gnu/glibc/)
3. Take glibc-2.37 as an example.
```
tar xvf glibc-2.37.tar.gz
cd glibc-2.37
mkdir build
cd ./build
../configure --prefix=<the path you want to install>
make -j<the number of your CPU cores> #if your cpu has 8 cores, then "make -j8"
sudo make install
sudo rm -rf /lib/x86_64-linux-gnu/libc.so.6
sudo ln -s <the path you installed glibc>/lib/libc-2.37.so /lib/x86_64-linux-gnu/libc.so.6
```

**Reply by PavanKumarMiriyala** [reply]
Yes, it is due to old libc version in my OS. I resolved that issue. Thank you so much.
